### PR TITLE
Install stack protection for TestWorkController

### LINKED ISSUE #4732: Testing subclasses of Task with redefined __call__ fails
## Checklist

- [ ] I have included the output of ``celery -A proj report`` in the issue.
      (if you are not able to do this, then at least specify the Celery
       version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.

## Steps to reproduce

1. Create subclass of Task with redefined ` __call__` which calls `super()`
2. Create bound task with this class
3. write tests with `celery_worker` fixture
4. `_task_stack` is empty

## Expected behavior
`_install_stack_protection` should be called in `TestWorkController.on_before_init` as well as in `WorkerController`
```python
from celery.app.trace import _install_stack_protection
from celery.contrib.testing.worker import TestWorkController

class ProtectedTestWorkController(TestWorkController):
    def on_before_init(self, quiet=False, **kwargs):
        # https://github.com/celery/celery/blob/v4.1.0/celery/app/trace.py#L606
        _install_stack_protection()
```

## Actual behavior
Sometimes it fails with random errors which are not saying anything about actual problem

### PR DESCRIPTION
**Pull request by imomaliev**
## Description
Fixes #4732

### CODE REVIEW

**Comment by imomaliev** [comment]
@auvipy How can I run integration tests locally? `docker-compose run --rm celery tox` taking too long, and `docker-compose run --rm celery tox -e 3.6-integration-redis` is just hangs and does nothing

File: celery/contrib/testing/worker.py [hunk 1]
```diff
@@ -130,6 +137,8 @@ def _start_worker_thread(app,
     state.should_terminate = 0
     t.join(10)
     state.should_terminate = None
+    # worker.hub.poller gets closed so we should reset _current_loop
+    worker.hub.reset()
```
Comment (imomaliev) [inline celery/contrib/testing/worker.py]: For some reason after worker is terminated _current_loop is still in memory, but doesn't have a poller anymore. We should either call `set_event_loop(None)` or `worker.hub.reset()`. Which one is preferred?

**Comment by imomaliev** [comment]
@auvipy For some reason after worker is terminated `_current_loop` is still in memory, but doesn't have a `poller` anymore. We have to call `set_event_loop(None)` or `worker.hub.reset()`. Which one is preferred?

**Comment by imomaliev** [comment]
@auvipy rebased

**Comment by auvipy** [comment]
I restarted the failed builds

**Comment by auvipy** [comment]
restarted the builds again. could you re base again please?

**Comment by imomaliev** [comment]
@auvipy rebased

**Comment by auvipy** [comment]
Waiting for the travis. it's damn slow

**Comment by imomaliev** [comment]
@auvipy it has been a long time since I made this PR but I will try to look into this next week

**Comment by imomaliev** [comment]
@auvipy rebased

**Comment by imomaliev** [comment]
> what do you think about the comment left on the ticket? and could you please rebase?

will look into this

**Comment by auvipy** [comment]
integration tests are failing

**Comment by auvipy** [comment]
can you revisit?

**Comment by imomaliev** [comment]
rebased

**Comment by auvipy** [comment]
restarted to check integration tests

**Comment by imomaliev** [comment]
I don't think increasing time will help, there is some other issue which makes integrations test to hang, I don't understand why. Probably because of `worker.hub.reset()` call

**Comment by auvipy** [comment]
at least can you try to fix the failing test case?

**Pull request closed without merging**